### PR TITLE
Doc: proposed Lido V2 addresses on mainnet

### PR DESCRIPTION
**Pull request by TheDZhon**
The proposed set of the Lido V2 addresses.

### CODE REVIEW

File: docs/deployed-contracts/mainnet-lido-v2-proposed.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# Proposed Lido V2 Addresses
+
+## Lido V2 Core Protocol
+
+- (proposed) LidoLocator: [`0xC1d0b3DE6792Bf6b4b37EccdcC24e45978Cfd2Eb`](https://etherscan.io/address/0xC1d0b3DE6792Bf6b4b37EccdcC24e45978Cfd2Eb) (proxy)
```
Comment (folkyatina) [inline docs/deployed-contracts/mainnet-lido-v2-proposed.md:5]: Maybe mark proxy explicitly like
```suggestion
- (proposed) LidoLocator proxy: [`0xC1d0b3DE6792Bf6b4b37EccdcC24e45978Cfd2Eb`](https://etherscan.io/address/0xC1d0b3DE6792Bf6b4b37EccdcC24e45978Cfd2Eb) (proxy)
```

File: docs/deployed-contracts/mainnet-lido-v2-proposed.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+- (proposed) Deposit Security module: [`0xC77F8768774E1c9244BEed705C4354f2113CFc09`](https://etherscan.io/address/0xC77F8768774E1c9244BEed705C4354f2113CFc09)
+- Execution Layer Rewards Vault: [`0x388C818CA8B9251b393131C08a736A67ccB19297`](https://etherscan.io/address/0x388C818CA8B9251b393131C08a736A67ccB19297)
+- (proposed) WithdrawalQueueERC721: [`0x889edC2eDab5f40e902b864aD4d7AdE8E412F9B1`](https://etherscan.io/address/0x889edC2eDab5f40e902b864aD4d7AdE8E412F9B1) (proxy)
+  - proposed implementation: [`0xE42C659Dc09109566720EA8b2De186c2Be7D94D9`](https://etherscan.io/address/0xE42C659Dc09109566720EA8b2De186c2Be7D94D9)
+- WithdrawalVault: [`0xb9d7934878b5fb9610b3fe8a5e441e8fad7e293f`](https://etherscan.io/address/0xb9d7934878b5fb9610b3fe8a5e441e8fad7e293f) (proxy)
+  - proposed implementation: [`0xCC52f17756C04bBa7E377716d7062fC36D7f69Fd`](https://etherscan.io/address/0xCC52f17756C04bBa7E377716d7062fC36D7f69Fd)
+- (proposed) Burner of stETH: [`0xD15a672319Cf0352560eE76d9e89eAB0889046D3`](https://etherscan.io/address/0xD15a672319Cf0352560eE76d9e89eAB0889046D3)
```
Comment (folkyatina) [inline docs/deployed-contracts/mainnet-lido-v2-proposed.md:21]: Strange to see `of stETH` here. Let's stick to contract names. Or add a short description separately if required.
```suggestion
- (proposed) Burner: [`0xD15a672319Cf0352560eE76d9e89eAB0889046D3`](https://etherscan.io/address/0xD15a672319Cf0352560eE76d9e89eAB0889046D3)
```

File: docs/deployed-contracts/mainnet-lido-v2-proposed.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+- MEV Boost Relay Allowed List: [`0xF95f069F9AD107938F6ba802a3da87892298610E`](https://etherscan.io/address/0xf95f069f9ad107938f6ba802a3da87892298610e)
+- Legacy Oracle (obsolete): [`0x442af784A788A5bd6F42A01Ebe9F287a871243fb`](https://etherscan.io/address/0x442af784A788A5bd6F42A01Ebe9F287a871243fb) (proxy)
```
Comment (folkyatina) [inline docs/deployed-contracts/mainnet-lido-v2-proposed.md:23]:
```suggestion
- LegacyOracle (LidoOracle before V2): [`0x442af784A788A5bd6F42A01Ebe9F287a871243fb`](https://etherscan.io/address/0x442af784A788A5bd6F42A01Ebe9F287a871243fb) (proxy)
```

**Review by folkyatina — COMMENTED** [review]
I checked addresses, but there is some thoughts about the format

Looks like sometimes we are lacking the description and having too many brackets

Let's try to unify the format. Something like this comes to my mind:
- [proposed ]Description (ContractName): [address]

So 
- (proposed) EIP-712 helper for stETH: [`0x8F73e4C2A6D852bb4ab2A45E6a9CF5715b3228B7`](https://etherscan.io/address/0x8F73e4C2A6D852bb4ab2A45E6a9CF5715b3228B7)
- (proposed) LidoLocator: [`0xC1d0b3DE6792Bf6b4b37EccdcC24e45978Cfd2Eb`](https://etherscan.io/address/0xC1d0b3DE6792Bf6b4b37EccdcC24e45978Cfd2Eb) (proxy)

will become:

- proposed EIP-712 helper for stETH (EIP712StETH): [`0x8F73e4C2A6D852bb4ab2A45E6a9CF5715b3228B7`](https://etherscan.io/address/0x8F73e4C2A6D852bb4ab2A45E6a9CF5715b3228B7)
- proposed LidoLocator proxy (OssifiableProxy): [`0xC1d0b3DE6792Bf6b4b37EccdcC24e45978Cfd2Eb`](https://etherscan.io/address/0xC1d0b3DE6792Bf6b4b37EccdcC24e45978Cfd2Eb)

**Review by folkyatina — APPROVED** [review]

**Review by zuzueeka — COMMENTED** [review]
1. There's no Gate Seal Blueprint in 
https://github.com/lidofinance/scripts/blob/shapella-upgrade/configs/config_shapella_addresses_mainnet.py
and
https://github.com/lidofinance/lido-dao/blob/feature/shapella-upgrade/config.json   

2. There's no Gate Seal Factory and Gate Seal in https://github.com/lidofinance/lido-dao/blob/feature/shapella-upgrade/config.json

3. There's no Shapella Upgrade Template in  https://github.com/lidofinance/lido-dao/blob/feature/shapella-upgrade/config.json

4. Do we need to add this address dummy_implementation_address = "0x6F6541C2203196fEeDd14CD2C09550dA1CbEDa31" ?

**Review by kadmil — APPROVED** [review]
LGTM

**Pull request merged**